### DISCUSSION TITLE
Fragments with inline HTML inside template tags

### DISCUSSION THREAD

**Discussion by thetutlage** [unpoly/unpoly#256] [Ideas]
This is more of a question out of curiosity. So feel free to lemme know, if this is against the goals of unpoly.

I was imaging, if it is possible to load the contents of a layer from an inline `template` tag vs making an HTTP request to the server.

The main use case is to display modal for confirmation messages. For example:

```html
    <div>
      <a href="#templateId" up-layer="new"> Delete post </a>
      <template id="templateId">
        <p>Contains the contents for the modal</p>
      </template>
    </div>
```

So when I will click the anchor tag, it will load the contents from the template tag vs making an HTTP request.

## Can't I achieve the same using standard JavaScript (outside of unpoly landscape)?

Yes, I can. But since unpoly modals are accessible. I will prefer not re-implementing the modal dialogs myself and rely on what's already there :)

**Comment by triskweline** [comment]
There are several ways to do this.

One way is to pass your modal's HTML as an `[up-content]` attribute:

```html
<a href="#templateId" up-layer="new" up-content="Contents for the modal">Delete post </a>
```

As an alternative you should also be able to implement the `<template>` scheme that you suggested. The following (untested) event handler will run when an `<a up-layer="new">` is clicked, and use local HTML from a `<template>` element if the link's `[href]` begins with an hash:

```js
up.on('up:link:follow', 'a[up-layer=new]', function({ renderOptions }) {
  if (renderOptions.url.startsWith('#')) {
    // Find the <template> with an ID matching the given URL.
    // E.g. if we're given { url: '#foo' } we will look for an element "template#foo".
    let template = document.querySelector("template" + renderOptions.url)
    renderOptions.url = undefined
    renderOptions.content = template.innerHTML
  }
})
```

Then use it as you did in your post:

```html
<a href="#templateId" up-layer="new"> Delete post </a>
<template id="templateId">
  <p>Contains the contents for the modal</p>
</template>
```

I didn't test this.

**Reply by thetutlage** [reply]
Hello @triskweline 

Thanks for the input. Yes, I am able to get it to work with one modification. An anchor without a URL (with just #) is ignored by unpoly on purpose. Code reference https://github.com/unpoly/unpoly/blob/master/src/unpoly/link.js#L188-L195. So I have to define an empty `up-content` as well.

```html
<a href="#templateId" up-layer="new" up-content> Delete post </a>
<template id="templateId">
  <p>Contains the contents for the modal</p>
</template>
```

**Reply by triskweline** [reply]
Ah yes, I forgot about that :+1:
